modules/core/04-channel/keeper/packet.go

```diff
@@ -40,22 +40,24 @@ func (k Keeper) SendPacket(
 		)
 	}
 
-	if !k.scopedKeeper.AuthenticateCapability(ctx, channelCap, host.ChannelCapabilityPath(packet.GetSourcePort(), packet.GetSourceChannel())) {
-		return sdkerrors.Wrapf(types.ErrChannelCapabilityNotFound, "caller does not own capability for channel, port ID (%s) channel ID (%s)", packet.GetSourcePort(), packet.GetSourceChannel())
+	if !k.scopedKeeper.AuthenticateCapability(ctx, channelCap, host.ChannelCapabilityPath(srcPort, srcChannel)) {
+		return sdkerrors.Wrapf(types.ErrChannelCapabilityNotFound, "caller does not own capability for channel, port ID (%s) channel ID (%s)", srcPort, srcChannel)
 	}
 
-	if packet.GetDestPort() != channel.Counterparty.PortId {
+	sequence, found := k.GetNextSequenceSend(ctx, srcPort, srcChannel)
```

Comment:
[q] Could you help me understand why this check is a substitute for the previous `packet.GetDestPort() != channel.Counterparty.PortId` and `packet.GetDestChannel() != channel.Counterparty.ChannelId`? It's not immediately clear to me looking into `GetNextSequenceSend`

Reply:
@crodriguezvega is correct. Yea the diffs on this PR look confusing in the merged view. But these checks have been removed completely since we are directly grabbing the destination info from the channelEnd

Before we were requiring callers to pass this in and checking it is the exact value we expect. But if we already have the value, then we could just plug it in ourselves. That is the point of this PR.

Reply:
Ah I see, thanks for the clarification 👍